src/main/java/duke/io/Storage.java

```diff
@@ -0,0 +1,103 @@
+import java.time.LocalTime;
+import java.util.Scanner;
+
+/**
+ * Represents the local persistent storage in Duke application.
+ *
+ * @author Zheng Teck
+ * @version 1.0
+ */
+public class Storage {
+
+    public static final String PATH = String.join(File.separator,"data","duke.txt");
+    private final File file;
+
+    /**
+     * Load the storage instance from local storage if exist else create a new instance.
+     */
+    public Storage() {
+        this.file = new File(PATH);
+        if (!this.file.getParentFile().exists())  {
+            this.file.getParentFile().mkdir();
+        }
+    }
+
+    /**
+     * This method writes data in the task list to local storage.
+     *
+     * @param tasks The list of task in the Duke application.
+     * @exception IOException
+     * @see IOException
+     */
+    public void writeToFile(TaskList tasks) throws IOException {
+        FileWriter fw = new FileWriter(PATH);
+        StringBuilder result = new StringBuilder();
+        for (Task task : tasks.getTaskList()) {
+            result.append(task.formatToSave()).append("\n");
+        }
+        fw.write(result.toString());
+        fw.close();
+    }
+
+    /**
+     * This method read the local storage to import tasks from the local persistent storage.
+     *
+     * @return Returns the imported list of task from the local persistent storage.
+     * @exception IOException
+     * @see IOException
+     */
+    public TaskList importTasks() throws IOException {
+        TaskList taskList = new TaskList();
+        try {
+            Scanner s = new Scanner(this.file);
+            String[] input;
+            LocalDate date;
+            LocalTime time;
+            while (s.hasNext()) {
+                String[] inputLine = s.nextLine().split("\\|");
+                String type = inputLine[0];
+                boolean isDone = inputLine[1].equals("1");
+                switch (type) {
+                    case "T":
+                        taskList.addToDo(inputLine[2],isDone);
+                        break;
+                    case "E":
+                        input = inputLine[3].split(" ");
+                        if (input.length != 2) {
+                            throw new DukeException(Ui.MSG_FILEREADERROR);
+                        }
+                        date = Parser.formatDate(input[0]);
+                        time = Parser.formatTime(date, input[1]);
+                        taskList.addEvent(inputLine[2],isDone, date, time);
+                        break;
+                    case "D":
+                        input = inputLine[3].split(" ");
+                        if (input.length != 2) {
+                            throw new DukeException(Ui.MSG_FILEREADERROR);
+                        }
+                        date = Parser.formatDate(input[0]);
+                        time = Parser.formatTime(date, input[1]);
+                        taskList.addDeadline(inputLine[2],date,time);
+                }
```

Comment:
Maybe to improve readability is to assign variables to inputLine[0/1/2/3]. I am guilty of this as well.

Reply:
Hi, I do not understand what do you mean by `assign variables to inputLine[0/1/2/3]`, could you please explain further. Cheers!

Reply:
I was confused in a way I have to make sense of what is input[3] or input[0] in different cases. Upon closer look, you did specify what is inputLine[0] by assigning it to type and date = input[0] etc. I think it is fine. Maybe not use reassign input but use dateTime to make it more readable?